core/chains/evm/gas/block_history_estimator.go

```diff
@@ -426,15 +425,18 @@ func (b *BlockHistoryEstimator) FetchBlocks(ctx context.Context, head *evmtypes.
 		return err
 	}
 
+	var missingBlocks []int64
 	for _, req := range reqs {
 		result, err := req.Result, req.Error
 		if err != nil {
-			if strings.Contains(err.Error(), "failed to decode block number while unmarshalling block") {
-				lggr.Errorw(
-					fmt.Sprintf("Failed to fetch block: RPC node returned an empty block on query for block number %d even though the WS subscription already sent us this block. It might help to increase BLOCK_HISTORY_ESTIMATOR_BLOCK_DELAY (currently %d)",
-						HexToInt64(req.Args[0]), blockDelay,
+			if errors.Is(err, ErrMissingBlock) {
+				num := HexToInt64(req.Args[0])
```

Comment:
Is it possible that `req.Args` can be empty?
Also it looks strange that `HexToInt64` does not return an error...

Reply:
No, it cannot be empty since it was populated just above. This code is the same as before, just moved.